script/validate-data/index.ts

```diff
@@ -104,10 +104,17 @@ async function checkWorkflow(workflowPath: string, propertiesPath: string, allow
       }
       
     }
+    var directoryName = dirname(workflowPath)
+    var folder_category = folder_category_map.find( folder_category => folder_category["name"] == directoryName)["category"]
     if (!workflowPath.endsWith("blank.yml") && (!properties.categories || 
       !properties.categories.some(category => allowed_categories.some(ac => ac.toLowerCase() == category.toLowerCase())))) {
       workflowErrors.errors.push(`Workflow does not contain at least one allowed category - ${allowed_categories}`)
     }
+
+    if(properties.categories && !properties.categories.some(category => category.toLowerCase() == folder_category.toLowerCase())) {
+      workflowErrors.errors.push(`Either workflow is not added to the correct directory or category specified is wrong. Allowed category for ${basename(directoryName)} directory is ${folder_category}`)
```

Comment:
```Allowed category for ${basename(directoryName)} directory is ${folder_category}```
This may be confusing as `categories` on properties.json contains language/tech-stacks as well. Call out that the first entry in `categories` should be _X_

Reply:
done